tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
+					 if(qbx > 70.){
+						qbx = qbx / 200.;
+						cout<<"XXXXXXX "<< qb << " , " << fTofdQ << " , "<<qbx<<std::endl;
+					 }
+			    }
+			    else
+			    {
+				    // double exponential
+				    auto pos = ((bot_ns + par->GetOffset1()) - (top_ns + par->GetOffset2())) * par->GetVeff();
+				    auto q1 = Cbot_tot /
+					    (parf[0] * (exp(-parf[1] * (pos + 100.)) + exp(-parf[2] * (pos + 100.))) + parf[3]*1.);
+				    parq[0] = par->GetPar2a();
+				    parq[1] = par->GetPar2b();
+				    parq[2] = par->GetPar2c();
+				    parq[3] = par->GetPar2d();
+							for(int i=0;i<4;i++){
+								cout<<"parf["<<i<<"] "<<parf[i]<<" parq["<<i<<"] "<<parq[i]<<endl;
+							}
+				    auto q2 = Ctop_tot /
+					    (parq[0] * (exp(-parq[1] * (pos + 100.)) + exp(-parq[2] * (pos + 100.))) + parq[3]*1.);
+				    q1 = q1 *
+					    fTofdQ / 200.; // theory says: dE ~ Z^2 but we see quenching -> just use linear and fit the rest
+				    q2 = q2 * fTofdQ / 200.;
+					cout<<"q1 "<<q1<<" q2 "<<q2<<endl;
+				    qb = (q1 + q2) / 2.;
+			    }
+
+				        std::cout<< " QB: " << posToT << " , " << qb << std::endl;
+			    // Fill control histograms and Q vs Pos without multihits
+			    
+				if(fParameter == 6){
+					Double_t parz[3];
+                parz[0] = par->GetPar1za();
+                parz[1] = par->GetPar1zb();
+                parz[2] = par->GetPar1zc();
+
+				qbx = parz[0] + parz[1]*qbx + parz[2]*qbx*qbx;
+				}
+				
+				if (multihits[iPlane - 1][iBar - 1] < 2 /*&& qb>190. */ || 1)
+			    {
+				    fhQvsPos[iPlane - 1][iBar - 1]->Fill(posToT, qb);
+					fhQXvsPos[iPlane - 1][iBar - 1]->Fill(posToT, qbx);
+			    }
+	
+			}
+	/*	    if(fParameter==5)
+		    {
+			    auto par = fHitPar->GetModuleParAt(iPlane, iBar);
+
+			    Double_t parz[3];
+			    parz[0] = par->GetPar1za();
+			    parz[1] = par->GetPar1zb();
+			    parz[2] = par->GetPar1zc();
+			    //parq[3] = par->GetPold();
+
+
+
+			    Double_t qb = 0.;
+			    if (fTofdSmiley)
+			    {
+				    qb = TMath::Sqrt(top_tot * bot_tot);///(parq[0] + parq[1] * posToT + parq[2] * pow(posToT, 2) + parq[3] * pow(posToT, 3));
+				    qb = qb *
+					    fTofdQ; // theory says: dE ~ Z^2 but we see quenching -> just use linear and fit the rest
+				    // std::cout<< " hola30 " << qb << " , " << fTofdQ << std::endl;
+			    }
+
+			    // Fill control histograms and Q vs Pos without multihits
+			    if (multihits[iPlane - 1][iBar - 1] < 2 && (qb > 5.6 && qb < 6.7.))
+			    {
+				    //    std::cout<< " h: " << posToT << " , " << qb << std::endl;
+				    fhZoffset[iPlane - 1][iBar - 1]->Fill(parz[0]+parz[1]*qb);
+			    }
+
+
+
+		    }
+*/
+		    ++top_i;
+		    ++bot_i;
+
+		    // Increment events
+		    fNEvents += 1;
+		}
+		else if (dt < 0 && dt > -c_range_ns / 2)
+		{
+			++top_i;
+		}
+		else
+		{
+			++bot_i;
+		}
+	    }
+	}
+    }
+	void R3BTofDCal2HitPar::SetHistogramsNULL(Int_t iPlane, Int_t iBar){
+		fhTdiff[iPlane - 1] = NULL;
```

Comment:
This could be done in the header using in the syntax:

TH1F *fhNameofHisto[NofPlanes][NofBars]{};